### PR TITLE
doc: release-notes-3.2: release notes for shell

### PR DESCRIPTION
**Pull request by jakub-uC**
Add release notes for the shell subsystem.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

### CODE REVIEW

File: doc/releases/release-notes-3.2.rst [hunk 1]
```diff
@@ -1499,6 +1499,12 @@ Libraries / Subsystems
     the storage instance used by the settings backed to store its records.
 
 * Shell
+  * Added new API function checking shell readiness: :c:func:`shell_ready`.
```
Comment (fabiobaltieri) [inline doc/releases/release-notes-3.2.rst:1502]: Needs a whitespace before the indentation change.

File: doc/releases/release-notes-3.2.rst [hunk 2]
```diff
@@ -1499,6 +1499,12 @@ Libraries / Subsystems
...
+  * Added option to control formatting of the logging timestamp.
+  * Added missing asserts to the shell api functions
```
Comment (fabiobaltieri) [inline doc/releases/release-notes-3.2.rst:1504]: nit: missing dot at end of sentence

File: doc/releases/release-notes-3.2.rst [hunk 3]
```diff
@@ -1499,6 +1499,12 @@ Libraries / Subsystems
...
+  * MQTT backend: bug fix to handle negative return value of the wait function.
+  * A new ``backends`` command that lists the name and number of active shell backends.
+  * Fix handling mandatory args with optional raw arg
```
Comment (fabiobaltieri) [inline doc/releases/release-notes-3.2.rst:1507]: nit: missing dot at end of sentence

Comment (fabiobaltieri) [reply doc/releases/release-notes-3.2.rst:1507]: also s/Fix/Fixed/

**Review by fabiobaltieri — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Pull request merged**